src/main/webapp/fetch_recipe.js

```diff
@@ -30,14 +30,38 @@ const jsonrecipes = [
     category: 'dinner'
     }
 ]
-function addRandomRecipe() {
-
-  // Pick a random recipe.
-  const jsonrecipe = jsonrecipes[Math.floor(Math.random() * jsonrecipes.length)];
-  const recipe = '\n \n Link: '+ jsonrecipe.link +
-                 '\n \n Title: Peanut Butter Popcorn ' + jsonrecipe.title +
-                 ' \n \n Likes: '+ jsonrecipe.likes +'\n \n Category: '+ jsonrecipe.category;
+/**
+ * Finds the recipe of the given title
+ * 
+ * @param {string} title title of the recipe
+ */
+function search(title) {
+  let recipe = NaN;
+  for (let i = 0; i < jsonrecipes.length; i++) {
+    if (jsonrecipes[i].title == title) {
+      recipe = jsonrecipes[i];
+      break;
+    }
+  }
+  if (recipe == NaN) {
+      this.console.log("title not in recipe list");
+  }
+  return recipe;
+}
+window.onload = function showRecipe(){
+  
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const title = urlParams.get('title');
+  
+  let recipe = search(title);
+  this.console.log(recipe);
```

Comment:
In the error case where `recipe` returns as `NaN`, the remainder of this function contains undefined behavior.

If recipe is `NaN` here, it's better to display the error to the user like "Oops! <Title of recipe> not found". Probably by changing what you put in `recipe-container` to be the error message instead